src/hex/shape.rs

```diff
@@ -45,12 +73,174 @@ impl<T: Clone> HexShape<T> {
+    ///
+    /// ```
+    /// use gridava::core::tile::Tile;
+    /// use gridava::hex::shape::*;
+    ///
+    /// /// Creates a line of size 1, 0-1 inclusive, and sets the tiles to Some(1)
+    /// let my_shape = HexShape::make_line(shapeargs!(1, 0, true), || Tile::new(Some(1)));
+    /// ```
+    pub fn make_line<F>(args: ShapeArgs, constructor: F) -> HexShape<T>
+    where
+        T: Clone + Default,
+        F: FnMut() -> Tile<T>,
+    {
+        // Working in local space
+        let vertex_a = axial!(0, 0);
+        let vertex_b = vertex_a.make_vector(args.size as i32, args.rot_dir);
+        Self::make_shape(&[vertex_a, vertex_b], args.square_bb, constructor)
+    }
+
+    /// Create a triangle shape.
+    ///
+    /// Given a size and direction, see [`ShapeArgs`], this will create a triangle.
+    ///
+    /// see [`Self::make_shape`] for more.
+    ///
+    /// ```
+    /// use gridava::core::tile::Tile;
+    /// use gridava::hex::shape::*;
+    ///
+    /// /// Creates a triangle of size 1, 0-1 inclusive, and sets the tiles to Some(1)
+    /// let my_shape = HexShape::make_triangle(shapeargs!(1, 0, true), || Tile::new(Some(1)));
+    /// ```
+    pub fn make_triangle<F>(args: ShapeArgs, constructor: F) -> HexShape<T>
+    where
+        T: Clone + Default,
+        F: FnMut() -> Tile<T>,
+    {
+        // Working in local space
+        let vertex_a = axial!(0, 0);
+        let vertex_b = vertex_a.make_vector(args.size as i32, args.rot_dir);
+        let vertex_c = vertex_a.make_vector(args.size as i32, args.rot_dir + 1);
+
+        Self::make_shape(&[vertex_a, vertex_b, vertex_c], args.square_bb, constructor)
+    }
+
+    /// Create a rhombus shape.
+    ///
+    /// Given a size and direction, see [`ShapeArgs`], this will create a rhombus.
+    ///
+    /// see [`Self::make_shape`] for more.
+    ///
+    /// ```
+    /// use gridava::core::tile::Tile;
+    /// use gridava::hex::shape::*;
+    ///
+    /// /// Creates a rhombus of size 1, 0-1 inclusive, and sets the tiles to Some(1)
+    /// let my_shape = HexShape::make_rhombus(shapeargs!(1, 0, true), || Tile::new(Some(1)));
+    /// ```
+    pub fn make_rhombus<F>(args: ShapeArgs, constructor: F) -> HexShape<T>
+    where
+        T: Clone + Default,
+        F: FnMut() -> Tile<T>,
+    {
+        // Working in local space
+        let vertex_a = axial!(0, 0);
+        let vertex_b = vertex_a.make_vector(args.size as i32, args.rot_dir);
+        let vertex_c = vertex_a.make_vector(args.size as i32, args.rot_dir + 1);
+        let vertex_d = vertex_b.make_vector(args.size as i32, args.rot_dir + 1);
+
+        Self::make_shape(
+            &[vertex_a, vertex_b, vertex_c, vertex_d],
+            args.square_bb,
+            constructor,
+        )
+    }
+
     /// Translate the shape.
     ///
     /// Mutates the transform of the shape.
     ///
     /// ```
-    /// use gridava::hex::coordinate::{Axial, axial};
+    /// use gridava::hex::coordinate::*;
```

Comment:
This is more concise, but I'm wondering which is more idiomatic Rust... since (while highly unlikely in this case) with wildcard imports there's always the possiblity of new features in the module being imported clobbering something in this file's namespace.